### PR TITLE
[user][test] Make tests consistent between endpoints

### PR DESCRIPTION
**Pull request by jaylees14**
- Move tests to the structure `<verb><service><type>`
- Make comments consistent between each test
- Make ordering of tests consistent between handlers
- Avoid names where we require the use of `a` or `an` depending on the name

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]

File: user/user_test.go [hunk 1]
```diff
@@ -127,19 +147,19 @@ func TestUserCreateHandlerFailsOnNoBody(t *testing.T) {
...
 
-	// Make a user
+	// Make a single user
 	_, err := makeRequest(mockEnv, http.MethodPost, "/user", `{"Name": "Jay"}`)
 	if err != nil {
 		t.Fatalf("Could not make POST request: %s", err.Error())
 	}
 
-	// Access that same user
+	// Read that same user
```
Comment (willjvsmith) [inline user/user_test.go]: Not a big deal, but would be nice if `// Create a single user`

File: user/user_test.go [hunk 2]
```diff
@@ -173,8 +193,8 @@ func TestUserReadHandlerFailsOnEmptyID(t *testing.T) {
 	}
 }
 
-// Test that providing an invalid ID to the read endpoint fails
-func TestUserReadHandlerFailsOnNonExistentID(t *testing.T) {
+// Test that providing a non existent ID to the read endpoint fails
```
Comment (willjvsmith) [inline user/user_test.go]:
```suggestion
// Test that providing a non-existent ID to the read endpoint fails
```

File: user/user_test.go [hunk 3]
```diff
@@ -293,14 +336,14 @@ func TestUserUpdateHandlerFailsOnEmptyID(t *testing.T) {
...
-	// 404 Not Found, since no route is defined for PUT at /user
+	// Not Found, since no route is defined for PUT at /user
 	if res.Code != http.StatusNotFound {
 		t.Errorf("Wrong status code %v", res.Code)
 	}
 }
 
-// Test that providing an invalid ID to the update endpoint fails
-func TestUserUpdateHandlerFailsOnNonExistentID(t *testing.T) {
+// Test that providing a non existent ID to the update endpoint fails
```
Comment (willjvsmith) [inline user/user_test.go]:
```suggestion
// Test that providing a non-existent ID to the update endpoint fails
```

File: user/user_test.go [hunk 4]
```diff
@@ -373,14 +416,14 @@ func TestUserDeleteHandlerFailsOnEmptyID(t *testing.T) {
...
-	// 404 Not Found, since no route is defined for DELETE at /user
+	// Not Found, since no route is defined for DELETE at /user
 	if res.Code != http.StatusNotFound {
 		t.Errorf("Wrong status code %v", res.Code)
 	}
 }
 
-// Test that providing an invalid ID to the delete endpoint fails
-func TestUserDeleteHandlerFailsOnNonExistentID(t *testing.T) {
+// Test that providing a non existent ID to the delete endpoint fails
```
Comment (willjvsmith) [inline user/user_test.go]:
```suggestion
// Test that providing a non-existent ID to the delete endpoint fails
```

**Review by willjvsmith — APPROVED** [review]
For reference, the test order we've followed here is:

* Succeeds
* Empty parameter
* Malformed body
* No body
* Empty ID
* Non-existent ID
* String ID

**Pull request merged**